src/librustc_ast/ast.rs

```diff
@@ -1904,6 +1904,7 @@ bitflags::bitflags! {
 pub enum InlineAsmTemplatePiece {
     String(String),
     Placeholder { operand_idx: usize, modifier: Option<char>, span: Span },
+    TemplateBoundary,
```

Comment:
Isn't this just a `String("\n")`? It would be nice not to have to deal with `TemplateBoundary` in the later stages of the compiler.

Reply:
Having a separate `TemplateBoundary` allows transforming inline assembly in the AST back to the exact set of template string arguments originally provided; I'd prefer to retain that property.

Reply:
We already don't preserve the template strings exactly since `{}` and `{name}` placeholders are turned into `{0}` in the AST. I don't think there is much value in preserving split template strings.

Reply:
@Amanieu Switching to a different approach that doesn't use `TemplateBoundary`, as discussed on Zulip.